lib/pbench/server/api/__init__.py

```diff
@@ -57,7 +57,8 @@ def register_endpoints(api, app, config):
 
     api.add_resource(
         DatasetsContents,
-        f"{base_uri}/datasets/contents/<string:dataset>",
+        f"{base_uri}/datasets/contents/<string:dataset>/",
+        f"{base_uri}/datasets/contents/<string:dataset>/<path:path>",
```

Comment:
If we are going to change from `parent` to `path`, why are we not also changing the `context` variable to `path` and the table of contents documents to `path`.  ToC entries have a keyword, `parent`, and a analyzed path, `directory`, to facilitate different ways to search.  I'd rather not introduce yet another name to add more confusion as to what goes where.

Reply:
I'm not concerned about the Elasticsearch mapping: we're just using ES as a hidden backend database, so (in theory) nobody outside the server needs to know how we organize data there. While "parent" may be, in that chained document structure, a convenient and even meaningful name for the "chain" field, I don't think it's a good representation of the "target" of `/datasets/contents`; our return payload is a JSON document that consolidates multiple ES documents. The caller is specifying a relative path within the tarball tree, and getting data about it.

Even for "contents" I felt that the name "parent" was ambiguous and a bit loose, but it's totally off the mark for "inventory", and I really wanted the two to be "twinsies". Whereas "path" seems like a good fit for both. We could say "target" but that seems less clear.

And, sure, changing the `CONTEXT` object key to be consistent is a good idea.

Reply:
Being consistent is good, except where we allow an internal representation to impact the external API without considering the consumer of the API.

In this case, I don't believe that is happening, though it seems we might be leaning towards having the API impact the internal representation.

When I read, `path`, I find that I ask myself: is `path` the full path including the `name` or is it the `path` to the parent?  `path` seems to be a name that indicates the form the contents take, but does not convey the semantic meaning.  I deliberately chose `parent` in the past because it conveys the meaning of the field, though I did not try to encode in the name the "type" of the value for that field (a directory path).

For the API, could we consider using `object` or `target` instead of `path`?

Reply:
While I see what you mean by "parent", that feels very unnatural to me; yes, `/datasets/contents` returns the contents or "children" of the named `path`/`target`, but if I saw "parent" in the documentation that's not what it implies to me. I guess because I'm pointing at the directory I want, not at the parent of the directory I want.

But `target` seems OK if that'll make everyone happy. Or, although I liked the idea of consistency between the two sibling APIs, we could be more explicit and call this one `directory` and the other `file` ...

Reply:
What's a "directory" (and "file")?  My storage system's API and GUI seems to have "folders" (and "objects").  😝

Anyway, 'target' is OK with me.